src/main/java/org/mule/extension/mulechain/helpers/AwsbedrockPayloadHelper.java

```diff
@@ -31,23 +33,63 @@
 import software.amazon.awssdk.services.bedrock.model.ListCustomModelsResponse;
 import software.amazon.awssdk.services.bedrock.model.ListFoundationModelsResponse;
 import software.amazon.awssdk.services.bedrock.model.ValidationException;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeClient;
+import software.amazon.awssdk.services.bedrockruntime.BedrockRuntimeClientBuilder;
+import org.mule.extension.mulechain.internal.CommonUtils;
+import org.mule.extension.mulechain.internal.TimeUnitEnum;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+
+
+
 
 public class AwsbedrockPayloadHelper {
 
     private static final Logger logger = LoggerFactory.getLogger(AwsbedrockPayloadHelper.class);
 
-    private static BedrockRuntimeClient createClient(AwsBasicCredentials awsCreds, Region region) {
-        return BedrockRuntimeClient.builder()
+    private static BedrockRuntimeClient createClient(AwsBasicCredentials awsCreds, Region region, AwsbedrockConfiguration configuration) {
+    	
+    	String endpointOverride = configuration.getEndpointOverride();
+    	
+        SdkHttpClient httpClient = CommonUtils.buildHttpClientWithTimeout(
+        	    configuration.getTimeout(),
+        	    configuration.getTimeoutUnit()
+        	);
+
+        
+     // Build the Bedrock client
+        BedrockRuntimeClientBuilder builder = BedrockRuntimeClient.builder()
                 .credentialsProvider(StaticCredentialsProvider.create(awsCreds))
                 .region(region)
-                .build();
+                .httpClient(httpClient);
+
+        if (endpointOverride != null && !endpointOverride.isEmpty()) {
+            builder.endpointOverride(URI.create(endpointOverride));
+        }
+
+        return builder.build();
+                
     }
+    
+    private static BedrockRuntimeClient createClientSession(AwsSessionCredentials awsCreds, Region region, AwsbedrockConfiguration configuration) {
+    	String endpointOverride = configuration.getEndpointOverride();
+    	
+        SdkHttpClient httpClient = CommonUtils.buildHttpClientWithTimeout(
+        	    configuration.getTimeout(),
+        	    configuration.getTimeoutUnit()
+        	);
+
 
-    private static BedrockRuntimeClient createClientSession(AwsSessionCredentials awsCreds, Region region) {
-        return BedrockRuntimeClient.builder()
+        BedrockRuntimeClientBuilder builder = BedrockRuntimeClient.builder()
                 .credentialsProvider(StaticCredentialsProvider.create(awsCreds))
                 .region(region)
-                .build();
+                .httpClient(httpClient);
+
+        if (endpointOverride != null && !endpointOverride.isEmpty()) {
```

Comment:
Generally speaking, it's better to check for `isBlank()` instead of `isEmpty()` ... the latter will cause failures if they accidentally enter a space for `endpointOverride`.